internal/handlers/common/filter.go

```diff
@@ -150,13 +163,18 @@ func filterOperator(doc *types.Document, operator string, filterValue any) (bool
 }
 
 // filterFieldExpr handles {field: {expr}} filter.
-func filterFieldExpr(fieldValue any, expr *types.Document) (bool, error) {
+func filterFieldExpr(fieldValue any, expr *types.Document, fieldExist bool) (bool, error) {
```

Comment:
`expr` and `fieldExist` describe the same thing there.

I think the right fix could be replace `fieldValue` with `field` and get `fieldValue` inside the function if needed. So, for example, if our document is `{"foo": "bar"}` and our query is `{"foo": {"$exists": true}}`, `fieldValue` would be "bar", but we really need "foo", so let's pass field itself instead of its value.

Reply:
I tried to do so but we actually need "bar" part almost everywhere to made comparison. So we can't just replace `fieldValue` with `filterKey`

Reply:
Let's pass the document, field and expr then. From document and field we can always get the value for that field in that document

Reply:
Yeah, I did it for ‘filterFieldExpr’, for the rest two cases I got the document value.